NDTensors/test/NDTensorsTestUtils/device_list.jl

```diff
@@ -13,7 +13,10 @@ if "rocm" in ARGS || "all" in ARGS
 end
 if "metal" in ARGS || "all" in ARGS
   ## Warning Metal does not work in Julia versions below 1.8
-  Pkg.add("Metal")
+  ## Essentially compat versioning here because we can't put Metal
+  ## in the Project.toml because it causes issues during testing in 
+  ## Julia versions below 1.8
+  Pkg.add(; name="Metal", version="1.1.0")
```

Comment:
@mtfishman since we don't have `Metal` in a the `Project.toml` of `NDTensors/test` this is a way to version control ourselves. I found that Jenkins was failing for metal because it was installing the wrong version. Adding this fixed it but the issue is that one will have to manually change this.
Another option would be to force an update on `Metal` and its dependencies with code like this 
```
Pkg.add("Metal")
Pkg.update("Metal"; preserve=PRESERVE_DIRECT)
```
Let me know what you think. Thanks!

Reply:
Do you know why it isn't installing the latest version in the first place? Also it seems fine to just call `Pkg.update()` if that works.

Reply:
I think it has something to do with this `preserve` variable. If I do something like this you can see Metal doesn't update properly.
```
Pkg.activate(temp=true)
Pkg.add(;name="Metal", version="1.0.0")
Pkg.update("Metal")
Updating registry at `~/.julia/registries/General.toml`
  No Changes to `/tmp/jl_XzMvJ2/Project.toml`
  No Changes to `/tmp/jl_XzMvJ2/Manifest.toml`
]st
Status `/tmp/jl_XzMvJ2/Project.toml`
⌃ [dde4c033] Metal v1.0.0
Info Packages marked with ⌃ have new versions available and may be upgradable.
```
But I don't really know why Pkg isn't grabbing the latest version of metal in the temporary project. Nowhere in our code do we explicitly compat Metal to 1.0.0 and the temporary package should always be clean.